### PR TITLE
Add yaml support.

### PR DESCRIPTION
**Pull request by dlorenc**
cc @nkubala 

This also starts us using Godeps for dependency management.

### CODE REVIEW

**Comment by dlorenc** [comment]
Fixes #23

File: structure_tests/structure_test.go [hunk 1]
```diff
@@ -36,48 +40,50 @@ type FileContentTest struct {
...
 		}
+		t.Logf("Executing: %s", cmd.Args)
 		var outbuf, errbuf bytes.Buffer
 
 		cmd.Stdout = &outbuf
 		cmd.Stderr = &errbuf
 
-		err := cmd.Run()
+		if err := cmd.Run(); err != nil {
+			// The test might be designed to run a command that exits with an error.
```
Comment (nkubala) [inline structure_tests/structure_test.go]: hmm ok. at first glance I was thinking we should add a `shouldFail` flag to the test, but I guess that can be implicit with the `expectedError` field.

File: structure_tests/structure_test.go [hunk 2]
```diff
@@ -36,48 +40,50 @@ type FileContentTest struct {
...
+		}
+		for _, errStr := range tt.ExcludedError {
+			errMsg := fmt.Sprintf("Excluded string '%s' found in error!", errStr)
+			compileAndRunRegex(errStr, stderr, t, errMsg, false)
 		}
 
 		for _, outStr := range tt.ExpectedOutput {
-			errMessage = "Expected string " + outStr + " not found in output!"
-			compileAndRunRegex(outStr, stdout, t, errMessage, true)
+			errMsg := fmt.Sprintf("Expected string '%s' not found in output!", outStr)
```
Comment (nkubala) [inline structure_tests/structure_test.go]: thanks :) was looking for the right way to format a string var

File: structure_tests/structure_test.go (hunk 1 above)
Comment (dlorenc) [inline structure_tests/structure_test.go]: Yeah I played around with a few different ways, but I think adding a field with the exit code is the best way. We can track that separately.

**Pull request merged**